vocabularies/DirectEdit.xml

```diff
@@ -0,0 +1,77 @@
+            <PropertyValue Property="href" String="https://sap.github.io/odata-vocabularies/vocabularies/DirectEdit.json" />
+          </Record>
+          <Record>
+            <PropertyValue Property="rel" String="describedby" />
+            <PropertyValue Property="href" String="https://github.com/sap/odata-vocabularies/blob/master/vocabularies/DirectEdit.md" />
+          </Record>
+        </Collection>
+      </Annotation>
+
+      <Term Name="SideEffects" Type="DirectEdit.SideEffectsType" Nullable="false" AppliesTo="EntitySet">
+        <Annotation Term="Common.Experimental" />
+        <Annotation Term="Core.Description" String="Determine side effects of client-side data modification" />
+      </Term>
+
+      <ComplexType Name="SideEffectsType">
+        <Annotation Term="Common.Experimental" />
+        <Annotation Term="Core.LongDescription">
+          <String>A direct-edit side-effect consists of:
+
+- a list of possible triggers, _preliminary_ syntax:
+  - `Path:` followed by a path to the property whose change triggers the side-effects determination 
+  - `CalculationAction:` followed by the name of the action that is called for the side-effects determination
+
+- an action bound to the entity set with the following non-binding parameters:
+  - `Qualifier` of type [`Core.SimpleIdentifier`](https://github.com/oasis-tcs/odata-vocabularies/blob/master/vocabularies/Org.OData.Core.V1.md#SimpleIdentifier)
+    or cast-compatible: the qualifier of the `SideEffects` annotation
+  - `Trigger` of type `Edm.String`: the trigger of side-effects determination, see `Triggers` property
+  - `Data` of either the entity type of the annotated entity set or a complex type that is structure-compatible with it, 
+  i.e.
+    - each primitive property that has the same name as a corresponding primitive property of the entity type of the annotated entity set is cast-compatible with the corresponding property and is nullable,
+    - each complex property that has the same name as a corresponding complex or navigation property of the entity type of the annotated entity set is structure-compatible with the corresponding property
+    - it may contain properties without a corresponding property in the entity type
+    - it may omit properties of the entity type
+  
+  The return type of the action also needs to be either the entity type of the annotated entity set or structure-compatible with it,
+  it can be the same type as for `Data`.</String>
+        </Annotation>
+        <Property Name="Triggers" Type="Collection(Edm.String)" Nullable="false">
+          <Annotation Term="Core.Description" String="List of possible triggers for determining side effects" />
+          <Annotation Term="Validation.MinItems" Int="1" />
+        </Property>
+        <Property Name="CalculationAction" Type="Common.QualifiedName" Nullable="false">
+          <Annotation Term="Core.Description" String="Name of the action for determining side effects" />
```

Comment:
```suggestion
        <Property Name="CalculationFunction" Type="Common.QualifiedName" Nullable="false">
          <Annotation Term="Core.Description" String="The operation calculating the side effects. While non-changing for the service, this technically is an action bound to the entity set so that the parameters can be sent in the POST request body. The action has the following non-binding parameters:
  - `Qualifier` of type [`Core.SimpleIdentifier`](https://github.com/oasis-tcs/odata-vocabularies/blob/master/vocabularies/Org.OData.Core.V1.md#SimpleIdentifier)
    or cast-compatible: the qualifier of the `SideEffects` annotation
  - `Trigger` of type `Edm.String`: the trigger of side-effects determination, see `Triggers` property
  - `Data` of either the entity type of the annotated entity set or a complex type that is structure-compatible with it, 
  i.e.
    - each primitive property that has the same name as a corresponding primitive property of the entity type of the annotated entity set is cast-compatible with the corresponding property and is nullable,
    - each complex property that has the same name as a corresponding complex or navigation property of the entity type of the annotated entity set is structure-compatible with the corresponding property
    - it may contain properties without a corresponding property in the entity type
    - it may omit properties of the entity type
  
  The return type of the action also needs to be either the entity type of the annotated entity set or structure-compatible with it,
  it can be the same type as for `Data`." />
```